clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -126,6 +141,12 @@ public synchronized void addTopics(String... topics) {
         requestUpdate();
     }
 
+    public synchronized void setTopics(Collection<String> topics) {
```

Comment:
I don't think it's possible since we await all pending requests to the coordinator before sending a new join group request, but I'll have to look into it. This is definitely one of the messier aspects of this patch. It may get easier after KAFKA-2388 since subscriptions are no longer additive, but the hard part is ensuring that the Metadata topic list matches the union of all topics subscribed by the group. One option might be to only update the Metadata topic list based on the subscriptions defined in the join group response. That would mean we'd always need two rounds of the rebalance when subscriptions change, which seems unfortunate. I'll see if there are any nicer options.